File: ALifeCommon/src/main/java/com/cristiano/alife/instructions/stack/PopInstruction.java
package com.cristiano.alife.instructions.stack;

import com.cristiano.alife.consts.ALifeConsts;
import com.cristiano.alife.world.BaseInstruction;
import com.cristiano.alife.world.IOrganismo;
import com.cristiano.alife.world.IWorld;

public class PopInstruction extends BaseInstruction  {
	public PopInstruction(IWorld mundo) {
		super(mundo);
	}
	@Override
	public void executa(IOrganismo o) {
		o.setReg(getByteOrganismo(o,1),o.pop());
	}

	@Override
	public String getDescription(IOrganismo o,int ip){
		return comment(name+" "+ALifeConsts.getLetter(o.getMemory(ip+1)),ALifeConsts.getLetter(o.getMemory(ip+1))+" <- pop()");
	}
	
	@Override
	public int getStep() {
		return 2;
	}
}

File: netAvida/netAvida.Backend/instructions/impl/stack/PopInstruction.cs
using netAvida.backend;
using netAvida.backend.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netAvida.Backend.instructions.impl
{
    public class PopInstruction : BaseInstruction
    {
        public PopInstruction(IWorld mundo): base(mundo)
        {
        }

    public override void executa(IOrganismo o)
        {
            o.setReg(getByteOrganismo(o, 1), o.pop());
        }

    public override int getStep()
        {
            return 2;
        }

    public override String getDescription(IOrganismo o, int ip)
        {
            return comment(name + " " + ALifeConsts.getLetter(o.getMemory(ip + 1)), ALifeConsts.getLetter(o.getMemory(ip + 1)) + " <- pop()");
        }
    }
}
